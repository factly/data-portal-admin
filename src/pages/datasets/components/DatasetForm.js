import React, { useState } from 'react';
import { Form, Input, InputNumber, Button, Select, notification } from 'antd';

import Selector from '../../../components/Selector';
import MediaUploader from '../../media/components/MediaUpload';

const { Option } = Select;

const formItemLayout = {
  labelCol: { span: 6 },
  wrapperCol: { span: 14 },
};

const DatasetForm = ({ onSubmit, setDatasetId, data, next }) => {
  const [media, setMedia] = useState(data?.feature_media);

  const formFields = [
    { label: 'Title', name: 'title', placeholder: 'Dataset title', required: true },
    {
      label: 'Contact Email',
      name: 'contact_email',
      placeholder: 'shashi@factly.in',
      required: true,
      type: 'email',
    },
    { label: 'Contact Name', name: 'contact_name', placeholder: 'Shashi', required: true },
    { label: 'Description', name: 'description', placeholder: 'description' },
    { label: 'Data Standard', name: 'data_standard', placeholder: 'standard' },
    { label: 'Price', type: 'number', name: 'price', placeholder: '1000' },
    { label: 'Currency', name: 'currency_id', placeholder: 'INR' },
    { label: 'Frequency', name: 'frequency', placeholder: '1 month', required: true },
    { label: 'Granularity', name: 'granularity', placeholder: 'granularity' },
    { label: 'License', name: 'license', placeholder: 'MIT License' },
    {
      label: 'Related Articles',
      name: 'related_articles',
      placeholder: 'link://to.related.articles',
    },
    { label: 'Source', name: 'source', placeholder: 'link://to.source.com' },
    { label: 'Temporal Coverage', name: 'temporal_coverage', placeholder: 'temporal_coverage' },
    { label: 'Time Saved', type: 'number', name: 'time_saved', placeholder: '12' },
  ];

  const onUploadSuccess = (media) => {
    setMedia(media);
  };

  const onFinish = (values) => {
    values.frequency = `${values.frequency.count} ${values.frequency.units}`;
    if (media) values.featured_media_id = media.id;

    onSubmit(values)
      .then((data) => {
        setDatasetId(data.id);
        notification.success({
          message: 'Success',
        });
        next();
      })
      .catch((error) => {
        console.log(error);
        notification.error({
          message: 'Error',
          description: 'Something went wrong',
        });
      });
  };

  return (
    <Form
      name="datasets_create_step_one"
      initialValues={data}
      {...formItemLayout}
      onFinish={onFinish}
    >
      {formFields.map((field) => (
        <Form.Item
          key={field.name}
          label={field.label}
          name={field.name}
          rules={
            field.required && [
              {
                required: true,
                message: `Please enter ${field.label}!`,
              },
            ]
          }
        >
          {field.type === 'number' ? (
            <InputNumber placeholder={field.placeholder} />
          ) : field.name === 'currency_id' ? (
            <Selector action="Currencies" field="iso_code" />
          ) : field.name === 'frequency' ? (
            <Input.Group compact>
              <Form.Item
                name={['frequency', 'count']}
                noStyle
                rules={[{ required: true, message: 'Frequency count is required' }]}
              >
                <InputNumber min={1} style={{ width: '15%' }} placeholder="12" />
              </Form.Item>
              <Form.Item
                name={['frequency', 'units']}
                noStyle
                rules={[{ required: true, message: 'Frequency unit is required' }]}
              >
                <Select placeholder="Select Unit">
                  <Option value="days">Days</Option>
                  <Option value="months">Months</Option>
                  <Option value="years">Years</Option>
                </Select>
              </Form.Item>
            </Input.Group>
          ) : (
            <Input placeholder={field.placeholder} type={field.type || ''} />
          )}
        </Form.Item>
      ))}
      <Form.Item key={'medium'} label={'Medium'}>
        {/* {media && media.name} */}
        <MediaUploader onUploadSuccess={onUploadSuccess} />
      </Form.Item>
      <Form.Item wrapperCol={{ span: 12, offset: 6 }}>
        <Button type="primary" htmlType="submit">
          Submit
        </Button>
      </Form.Item>
    </Form>
  );
};

export default DatasetForm;
