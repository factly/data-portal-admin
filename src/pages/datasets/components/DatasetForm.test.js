import React from 'react';
import { useDispatch } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { act } from '@testing-library/react';
import { mount, shallow } from 'enzyme';

import '../../../matchMedia.mock';
import DatasetForm from './DatasetForm';

const data = {
  id: 1,
  title: 'title',
  contact_email: 'contact_email',
  contact_name: 'contact_name',
  data_standard: 'data_standard',
  description: 'description',
  frequency: { count: 1, units: 'units' },
  granularity: 'granularity',
  license: 'license',
  related_articles: 'related_articles',
  source: 'source',
  temporal_coverage: 'temporal_coverage',
  time_saved: 2,
  feature_media: { id: 10, media: 'media' },
};

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: jest.fn(),
}));

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

let onSubmit, store;

describe('DatasetForm component', () => {
  store = mockStore({
    datasets: {
      loading: false,
      ids: [],
      req: [],
      items: {},
      total: 0,
    },
  });
  useDispatch.mockReturnValue(jest.fn());

  describe('snapshot testing', () => {
    beforeEach(() => {
      onSubmit = jest.fn();
      onSubmit.mockImplementationOnce(
        (values) => new Promise((resolve, reject) => resolve(values)),
      );
    });
    it('should render the component', () => {
      let tree;
      act(() => {
        tree = shallow(<DatasetForm />);
      });
      expect(tree).toMatchSnapshot();
      tree.unmount();
    });
    it('should match component in all steps', () => {
      let tree;
      act(() => {
        tree = shallow(<DatasetForm data={data} />);
      });
      expect(tree).toMatchSnapshot();
      tree.unmount();
    });
    it('should match component with data', () => {
      let tree;
      act(() => {
        tree = shallow(<DatasetForm onSubmit={onSubmit} data={data} />);
      });
      expect(tree).toMatchSnapshot();
    });
  });
  describe('component testing', () => {
    let wrapper, props;
    beforeEach(() => {
      const onSubmit = jest.fn();
      onSubmit.mockResolvedValueOnce({});

      props = {
        onSubmit: onSubmit,
        setDatasetId: jest.fn(),
        next: jest.fn(),
        data: data,
      };
      act(() => {
        wrapper = mount(<DatasetForm {...props} />);
      });
    });
    afterEach(() => {
      wrapper.unmount();
    });
    it('should not submit form with empty data', (done) => {
      act(() => {
        wrapper = mount(<DatasetForm onSubmit={props.onSubmit} />);
      });

      act(() => {
        const submitButtom = wrapper.find('Button').at(0);
        submitButtom.simulate('submit');
      });
      wrapper.update();

      setTimeout(() => {
        expect(props.onSubmit).not.toHaveBeenCalled();
        done();
      }, 0);
    });
    it('should submit form with given data', (done) => {
      act(() => {
        const submitButtom = wrapper.find('Button').at(0);
        submitButtom.simulate('submit');
      });
      wrapper.update();

      setTimeout(() => {
        expect(props.onSubmit).toHaveBeenCalledTimes(1);
        expect(props.onSubmit).toHaveBeenCalledWith({
          title: 'title',
          contact_email: 'contact_email',
          contact_name: 'contact_name',
          data_standard: 'data_standard',
          description: 'description',
          frequency: '1 units',
          granularity: 'granularity',
          license: 'license',
          related_articles: 'related_articles',
          source: 'source',
          temporal_coverage: 'temporal_coverage',
          time_saved: 2,
          featured_media_id: 10,
        });
        done();
      }, 0);
    });
    it('should submit form with given data without media', (done) => {
      act(() => {
        wrapper = mount(
          <DatasetForm
            onSubmit={props.onSubmit}
            data={{
              id: 1,
              title: 'title',
              contact_email: 'contact_email',
              contact_name: 'contact_name',
              data_standard: 'data_standard',
              description: 'description',
              frequency: { count: 1, units: 'units' },
              granularity: 'granularity',
              license: 'license',
              related_articles: 'related_articles',
              source: 'source',
              temporal_coverage: 'temporal_coverage',
              time_saved: 2,
            }}
          />,
        );
      });

      act(() => {
        const submitButtom = wrapper.find('Button').at(0);
        submitButtom.simulate('submit');
      });
      wrapper.update();

      setTimeout(() => {
        expect(props.onSubmit).toHaveBeenCalledTimes(1);
        expect(props.onSubmit).toHaveBeenCalledWith({
          title: 'title',
          contact_email: 'contact_email',
          contact_name: 'contact_name',
          data_standard: 'data_standard',
          description: 'description',
          frequency: '1 units',
          granularity: 'granularity',
          license: 'license',
          related_articles: 'related_articles',
          source: 'source',
          temporal_coverage: 'temporal_coverage',
          time_saved: 2,
        });
        done();
      }, 0);
    });
    it('should submit form with updated data', (done) => {
      act(() => {
        wrapper
          .find('FormItem')
          .at(0)
          .find('Input')
          .simulate('change', { target: { value: 'new title' } });
        wrapper
          .find('FormItem')
          .at(1)
          .find('Input')
          .simulate('change', { target: { value: 'new contact_email' } });
        wrapper
          .find('FormItem')
          .at(2)
          .find('Input')
          .simulate('change', { target: { value: 'new contact_name' } });
        wrapper
          .find('FormItem')
          .at(3)
          .find('Input')
          .simulate('change', { target: { value: 'new description' } });
        wrapper
          .find('FormItem')
          .at(4)
          .find('Input')
          .simulate('change', { target: { value: 'new data_standard' } });
        wrapper
          .find('FormItem')
          .at(5)
          .find('InputNumber')
          .at(0)
          .props()
          .onChange({ target: { value: 2 } });
        wrapper
          .find('FormItem')
          .at(5)
          .find('Select')
          .at(0)
          .props()
          .onChange({ target: { value: 'months' } });
        wrapper
          .find('FormItem')
          .at(6 + 2)
          .find('Input')
          .simulate('change', { target: { value: 'new granularity' } });
        wrapper
          .find('FormItem')
          .at(7 + 2)
          .find('Input')
          .simulate('change', { target: { value: 'new license' } });
        wrapper
          .find('FormItem')
          .at(8 + 2)
          .find('Input')
          .simulate('change', { target: { value: 'new related_articles' } });
        wrapper
          .find('FormItem')
          .at(9 + 2)
          .find('Input')
          .simulate('change', { target: { value: 'new source' } });
        wrapper
          .find('FormItem')
          .at(10 + 2)
          .find('Input')
          .simulate('change', { target: { value: 'new temporal_coverage' } });
        wrapper
          .find('FormItem')
          .at(11 + 2)
          .find('InputNumber')
          .at(0)
          .props()
          .onChange({ target: { value: 4 } });
        wrapper
          .find('FormItem')
          .at(12 + 2)
          .find('MediaUploader')
          .props()
          .onUploadSuccess({ id: 20, media: 'media' });

        const submitButtom = wrapper.find('Button').at(0);
        submitButtom.simulate('submit');
        wrapper.update();
      });

      setTimeout(() => {
        expect(props.onSubmit).toHaveBeenCalledTimes(1);
        expect(props.onSubmit).toHaveBeenCalledWith({
          title: 'new title',
          contact_email: 'new contact_email',
          contact_name: 'new contact_name',
          data_standard: 'new data_standard',
          description: 'new description',
          frequency: '2 months',
          granularity: 'new granularity',
          license: 'new license',
          related_articles: 'new related_articles',
          source: 'new source',
          temporal_coverage: 'new temporal_coverage',
          time_saved: 4,
          featured_media_id: 20,
        });
        done();
      }, 0);
    });
  });
});
