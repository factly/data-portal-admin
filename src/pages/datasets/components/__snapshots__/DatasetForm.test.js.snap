// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatasetForm component snapshot testing should match component in all steps 1`] = `
<ForwardRef(InternalForm)
  initialValues={
    Object {
      "contact_email": "contact_email",
      "contact_name": "contact_name",
      "data_standard": "data_standard",
      "description": "description",
      "feature_media": Object {
        "id": 10,
        "media": "media",
      },
      "frequency": Object {
        "count": 1,
        "units": "units",
      },
      "granularity": "granularity",
      "id": 1,
      "license": "license",
      "related_articles": "related_articles",
      "source": "source",
      "temporal_coverage": "temporal_coverage",
      "time_saved": 2,
      "title": "title",
    }
  }
  labelCol={
    Object {
      "span": 6,
    }
  }
  name="datasets_create_step_one"
  onFinish={[Function]}
  wrapperCol={
    Object {
      "span": 14,
    }
  }
>
  <FormItem
    key="title"
    label="Title"
    name="title"
    rules={
      Array [
        Object {
          "message": "Please enter Title!",
          "required": true,
        },
      ]
    }
  >
    <Input
      placeholder="Dataset title"
      type=""
    />
  </FormItem>
  <FormItem
    key="contact_email"
    label="Contact Email"
    name="contact_email"
    rules={
      Array [
        Object {
          "message": "Please enter Contact Email!",
          "required": true,
        },
      ]
    }
  >
    <Input
      placeholder="shashi@factly.in"
      type="email"
    />
  </FormItem>
  <FormItem
    key="contact_name"
    label="Contact Name"
    name="contact_name"
    rules={
      Array [
        Object {
          "message": "Please enter Contact Name!",
          "required": true,
        },
      ]
    }
  >
    <Input
      placeholder="Shashi"
      type=""
    />
  </FormItem>
  <FormItem
    key="description"
    label="Description"
    name="description"
  >
    <Input
      placeholder="description"
      type=""
    />
  </FormItem>
  <FormItem
    key="data_standard"
    label="Data Standard"
    name="data_standard"
  >
    <Input
      placeholder="standard"
      type=""
    />
  </FormItem>
  <FormItem
    key="frequency"
    label="Frequency"
    name="frequency"
    rules={
      Array [
        Object {
          "message": "Please enter Frequency!",
          "required": true,
        },
      ]
    }
  >
    <Group
      compact={true}
    >
      <FormItem
        name={
          Array [
            "frequency",
            "count",
          ]
        }
        noStyle={true}
        rules={
          Array [
            Object {
              "message": "Frequency count is required",
              "required": true,
            },
          ]
        }
      >
        <ForwardRef
          min={1}
          placeholder="12"
          step={1}
          style={
            Object {
              "width": "15%",
            }
          }
        />
      </FormItem>
      <FormItem
        name={
          Array [
            "frequency",
            "units",
          ]
        }
        noStyle={true}
        rules={
          Array [
            Object {
              "message": "Frequency unit is required",
              "required": true,
            },
          ]
        }
      >
        <Select
          bordered={true}
          choiceTransitionName=""
          placeholder="Select Unit"
          transitionName="slide-up"
        >
          <Option
            value="days"
          >
            Days
          </Option>
          <Option
            value="months"
          >
            Months
          </Option>
          <Option
            value="years"
          >
            Years
          </Option>
        </Select>
      </FormItem>
    </Group>
  </FormItem>
  <FormItem
    key="granularity"
    label="Granularity"
    name="granularity"
  >
    <Input
      placeholder="granularity"
      type=""
    />
  </FormItem>
  <FormItem
    key="license"
    label="License"
    name="license"
  >
    <Input
      placeholder="MIT License"
      type=""
    />
  </FormItem>
  <FormItem
    key="related_articles"
    label="Related Articles"
    name="related_articles"
  >
    <Input
      placeholder="link://to.related.articles"
      type=""
    />
  </FormItem>
  <FormItem
    key="source"
    label="Source"
    name="source"
  >
    <Input
      placeholder="link://to.source.com"
      type=""
    />
  </FormItem>
  <FormItem
    key="temporal_coverage"
    label="Temporal Coverage"
    name="temporal_coverage"
  >
    <Input
      placeholder="temporal_coverage"
      type=""
    />
  </FormItem>
  <FormItem
    key="time_saved"
    label="Time Saved"
    name="time_saved"
  >
    <ForwardRef
      placeholder="12"
      step={1}
    />
  </FormItem>
  <FormItem
    key="medium"
    label="Medium"
  >
    <MediaUploader
      onUploadSuccess={[Function]}
    />
  </FormItem>
  <FormItem
    wrapperCol={
      Object {
        "offset": 6,
        "span": 12,
      }
    }
  >
    <Button
      block={false}
      ghost={false}
      htmlType="submit"
      loading={false}
      type="primary"
    >
      Submit
    </Button>
  </FormItem>
</ForwardRef(InternalForm)>
`;

exports[`DatasetForm component snapshot testing should match component with data 1`] = `
<ForwardRef(InternalForm)
  initialValues={
    Object {
      "contact_email": "contact_email",
      "contact_name": "contact_name",
      "data_standard": "data_standard",
      "description": "description",
      "feature_media": Object {
        "id": 10,
        "media": "media",
      },
      "frequency": Object {
        "count": 1,
        "units": "units",
      },
      "granularity": "granularity",
      "id": 1,
      "license": "license",
      "related_articles": "related_articles",
      "source": "source",
      "temporal_coverage": "temporal_coverage",
      "time_saved": 2,
      "title": "title",
    }
  }
  labelCol={
    Object {
      "span": 6,
    }
  }
  name="datasets_create_step_one"
  onFinish={[Function]}
  wrapperCol={
    Object {
      "span": 14,
    }
  }
>
  <FormItem
    key="title"
    label="Title"
    name="title"
    rules={
      Array [
        Object {
          "message": "Please enter Title!",
          "required": true,
        },
      ]
    }
  >
    <Input
      placeholder="Dataset title"
      type=""
    />
  </FormItem>
  <FormItem
    key="contact_email"
    label="Contact Email"
    name="contact_email"
    rules={
      Array [
        Object {
          "message": "Please enter Contact Email!",
          "required": true,
        },
      ]
    }
  >
    <Input
      placeholder="shashi@factly.in"
      type="email"
    />
  </FormItem>
  <FormItem
    key="contact_name"
    label="Contact Name"
    name="contact_name"
    rules={
      Array [
        Object {
          "message": "Please enter Contact Name!",
          "required": true,
        },
      ]
    }
  >
    <Input
      placeholder="Shashi"
      type=""
    />
  </FormItem>
  <FormItem
    key="description"
    label="Description"
    name="description"
  >
    <Input
      placeholder="description"
      type=""
    />
  </FormItem>
  <FormItem
    key="data_standard"
    label="Data Standard"
    name="data_standard"
  >
    <Input
      placeholder="standard"
      type=""
    />
  </FormItem>
  <FormItem
    key="frequency"
    label="Frequency"
    name="frequency"
    rules={
      Array [
        Object {
          "message": "Please enter Frequency!",
          "required": true,
        },
      ]
    }
  >
    <Group
      compact={true}
    >
      <FormItem
        name={
          Array [
            "frequency",
            "count",
          ]
        }
        noStyle={true}
        rules={
          Array [
            Object {
              "message": "Frequency count is required",
              "required": true,
            },
          ]
        }
      >
        <ForwardRef
          min={1}
          placeholder="12"
          step={1}
          style={
            Object {
              "width": "15%",
            }
          }
        />
      </FormItem>
      <FormItem
        name={
          Array [
            "frequency",
            "units",
          ]
        }
        noStyle={true}
        rules={
          Array [
            Object {
              "message": "Frequency unit is required",
              "required": true,
            },
          ]
        }
      >
        <Select
          bordered={true}
          choiceTransitionName=""
          placeholder="Select Unit"
          transitionName="slide-up"
        >
          <Option
            value="days"
          >
            Days
          </Option>
          <Option
            value="months"
          >
            Months
          </Option>
          <Option
            value="years"
          >
            Years
          </Option>
        </Select>
      </FormItem>
    </Group>
  </FormItem>
  <FormItem
    key="granularity"
    label="Granularity"
    name="granularity"
  >
    <Input
      placeholder="granularity"
      type=""
    />
  </FormItem>
  <FormItem
    key="license"
    label="License"
    name="license"
  >
    <Input
      placeholder="MIT License"
      type=""
    />
  </FormItem>
  <FormItem
    key="related_articles"
    label="Related Articles"
    name="related_articles"
  >
    <Input
      placeholder="link://to.related.articles"
      type=""
    />
  </FormItem>
  <FormItem
    key="source"
    label="Source"
    name="source"
  >
    <Input
      placeholder="link://to.source.com"
      type=""
    />
  </FormItem>
  <FormItem
    key="temporal_coverage"
    label="Temporal Coverage"
    name="temporal_coverage"
  >
    <Input
      placeholder="temporal_coverage"
      type=""
    />
  </FormItem>
  <FormItem
    key="time_saved"
    label="Time Saved"
    name="time_saved"
  >
    <ForwardRef
      placeholder="12"
      step={1}
    />
  </FormItem>
  <FormItem
    key="medium"
    label="Medium"
  >
    <MediaUploader
      onUploadSuccess={[Function]}
    />
  </FormItem>
  <FormItem
    wrapperCol={
      Object {
        "offset": 6,
        "span": 12,
      }
    }
  >
    <Button
      block={false}
      ghost={false}
      htmlType="submit"
      loading={false}
      type="primary"
    >
      Submit
    </Button>
  </FormItem>
</ForwardRef(InternalForm)>
`;

exports[`DatasetForm component snapshot testing should render the component 1`] = `
<ForwardRef(InternalForm)
  labelCol={
    Object {
      "span": 6,
    }
  }
  name="datasets_create_step_one"
  onFinish={[Function]}
  wrapperCol={
    Object {
      "span": 14,
    }
  }
>
  <FormItem
    key="title"
    label="Title"
    name="title"
    rules={
      Array [
        Object {
          "message": "Please enter Title!",
          "required": true,
        },
      ]
    }
  >
    <Input
      placeholder="Dataset title"
      type=""
    />
  </FormItem>
  <FormItem
    key="contact_email"
    label="Contact Email"
    name="contact_email"
    rules={
      Array [
        Object {
          "message": "Please enter Contact Email!",
          "required": true,
        },
      ]
    }
  >
    <Input
      placeholder="shashi@factly.in"
      type="email"
    />
  </FormItem>
  <FormItem
    key="contact_name"
    label="Contact Name"
    name="contact_name"
    rules={
      Array [
        Object {
          "message": "Please enter Contact Name!",
          "required": true,
        },
      ]
    }
  >
    <Input
      placeholder="Shashi"
      type=""
    />
  </FormItem>
  <FormItem
    key="description"
    label="Description"
    name="description"
  >
    <Input
      placeholder="description"
      type=""
    />
  </FormItem>
  <FormItem
    key="data_standard"
    label="Data Standard"
    name="data_standard"
  >
    <Input
      placeholder="standard"
      type=""
    />
  </FormItem>
  <FormItem
    key="frequency"
    label="Frequency"
    name="frequency"
    rules={
      Array [
        Object {
          "message": "Please enter Frequency!",
          "required": true,
        },
      ]
    }
  >
    <Group
      compact={true}
    >
      <FormItem
        name={
          Array [
            "frequency",
            "count",
          ]
        }
        noStyle={true}
        rules={
          Array [
            Object {
              "message": "Frequency count is required",
              "required": true,
            },
          ]
        }
      >
        <ForwardRef
          min={1}
          placeholder="12"
          step={1}
          style={
            Object {
              "width": "15%",
            }
          }
        />
      </FormItem>
      <FormItem
        name={
          Array [
            "frequency",
            "units",
          ]
        }
        noStyle={true}
        rules={
          Array [
            Object {
              "message": "Frequency unit is required",
              "required": true,
            },
          ]
        }
      >
        <Select
          bordered={true}
          choiceTransitionName=""
          placeholder="Select Unit"
          transitionName="slide-up"
        >
          <Option
            value="days"
          >
            Days
          </Option>
          <Option
            value="months"
          >
            Months
          </Option>
          <Option
            value="years"
          >
            Years
          </Option>
        </Select>
      </FormItem>
    </Group>
  </FormItem>
  <FormItem
    key="granularity"
    label="Granularity"
    name="granularity"
  >
    <Input
      placeholder="granularity"
      type=""
    />
  </FormItem>
  <FormItem
    key="license"
    label="License"
    name="license"
  >
    <Input
      placeholder="MIT License"
      type=""
    />
  </FormItem>
  <FormItem
    key="related_articles"
    label="Related Articles"
    name="related_articles"
  >
    <Input
      placeholder="link://to.related.articles"
      type=""
    />
  </FormItem>
  <FormItem
    key="source"
    label="Source"
    name="source"
  >
    <Input
      placeholder="link://to.source.com"
      type=""
    />
  </FormItem>
  <FormItem
    key="temporal_coverage"
    label="Temporal Coverage"
    name="temporal_coverage"
  >
    <Input
      placeholder="temporal_coverage"
      type=""
    />
  </FormItem>
  <FormItem
    key="time_saved"
    label="Time Saved"
    name="time_saved"
  >
    <ForwardRef
      placeholder="12"
      step={1}
    />
  </FormItem>
  <FormItem
    key="medium"
    label="Medium"
  >
    <MediaUploader
      onUploadSuccess={[Function]}
    />
  </FormItem>
  <FormItem
    wrapperCol={
      Object {
        "offset": 6,
        "span": 12,
      }
    }
  >
    <Button
      block={false}
      ghost={false}
      htmlType="submit"
      loading={false}
      type="primary"
    >
      Submit
    </Button>
  </FormItem>
</ForwardRef(InternalForm)>
`;
